_sources/Unit1-Using-Objects-and-Methods/topic-1-7-APIs-and-libraries.rst

```diff
@@ -1,10 +1,202 @@
+    :figclass: align-center
+
+**Attributes** (or **fields**) refer to the data related to the class and are stored in variables. **Behaviors** refer to what instances of the class can do (or what can be done with it) and are defined by methods. We will learn more about these in the next lessons.
+
+The ``System`` class has three fields/attributes: ``in``, ``out``, and ``err``. The ``out`` attribute is the output stream we use in ``System.out.println()`` and an instance of the ``PrintStream`` class, documented at https://docs.oracle.com/javase/8/docs/api/java/io/PrintStream.html. The ``PrintStream`` class has the method ``println()`` that we use to print to the screen. How many ``println()`` methods are there in the PrintStream class?  One for each type that it can print!
+
+How can you tell attributes and methods apart when you look at library documentation? One easy way is that methods will always have parentheses after them, for example ``println()``, where you can put in the information (the arguments) that it needs to do its job. Attributes are variables and will not have parentheses after them. In later lessons, you will learn more about arguments and return values of methods, and you will learn to write your own classes and methods.
+
+|Groupwork| Challenge  
+------------------------
+
+Work in pairs (or independently if instructed by your teacher) to answer the following questions about APIs and libraries referring to https://docs.oracle.com/javase/8/docs/api/java/lang/System.html and https://docs.oracle.com/javase/8/docs/api/java/io/PrintStream.html. Make sure that you can identify the attributes and behaviors of a class found in the libraries contained in an API.
+
+|Exercise| **Check Your Understanding**
+
+
+.. dragndrop:: API_vocab_match
+    :feedback: Review the summaries above.
+    :match_1: A collection of classes written by other programmers|||library
+    :match_2: A collection of related classes and interfaces that can be imported into your code|||package
+    :match_3: A specification that tells the programmer how to use classes in a library|||API
+    :match_4: The building blocks of Java and object-oriented programming that create a reference type|||class
+    :match_5: A block of code that performs a specific task|||method
+    :match_6: Data related to the class stored in variables|||attribute
+
+    Drag the definition from the left and drop it on the correct concept on the right. Click the "Check Me" button to see if you are correct.
+
+.. mchoice:: built_in_package
+    :answer_a: java.util
+    :answer_b: java.io
+    :answer_c: java.lang
+    :answer_d: java.net
+    :correct: c
+    :feedback_a: Incorrect. java.util is not built into Java.
+    :feedback_b: Incorrect. java.io is not built into Java.
+    :feedback_c: Correct! java.lang is built into Java.
+    :feedback_d: Incorrect. java.net is not built into Java.
+
+    Which package is already built into Java and does not need to be imported?
```

Comment:
Hmmmm. I would say these are all "built into Java". However it is correct that they do not need to be imported. Maybe "Which package is implicitly imported into all Java programs."?